### PR TITLE
Yarn upgrade

### PR DESCRIPTION
**Pull request by Nicholassully**
"webpack": "^5.50.0",
"webpack-cli": "^4.7.2"
"cypress": "8.2.0"

Also fixed one failing test as was checking cypress upgrade

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -47,8 +47,8 @@
     "sync-request": "^6.0.0",
     "universal-analytics": "^0.4.16",
     "uuid": "^8.3.0",
-    "webpack": "^4.44.2",
-    "webpack-cli": "^3.3.12"
+    "webpack": "^5.50.0",
```
Comment (mattmachell) [inline package.json:50]: Going from Webpack 4 to 5 had some config changes that broke when I did it for Sirius Supervision. Presumably the build works here?

File: cypress/integration/filter.spec.js [hunk 2]
```diff
@@ -16,7 +16,7 @@ describe("Reassign Tasks", () => {
   it("can hide the filters", () => {
     cy.get('#option-select-title-task-type').click()
     cy.get('#option-select-title-task-type').click()
-    cy.get('app-c-option-select__container-inner').should('not.be.visible') 
+    cy.get('.govuk-fieldset > :nth-child(1) > .govuk-checkboxes__item > .govuk-label').should('not.be.visible')
```
Comment (mattmachell) [inline cypress/integration/filter.spec.js:19]: In general I'd avoid using visual styling hooks for test hooks, as this makes the test brittle to change that doesn't affect whether the interaction works breaking the test. 

One option is if you have label that should not be visible, better to identify it as specifically as possible (say by the for attribute for the form element it hooks onto).

That's all assuming you aren't deliberately testing the html structure, of course.

File: package.json [hunk 3]
```diff
@@ -20,7 +20,7 @@
     "client-sessions": "^0.8.0",
     "cookie-parser": "^1.4.5",
     "cross-spawn": "^7.0.2",
-    "cypress": "5.3.0",
+    "cypress": "8.2.0",
```
Comment (mattmachell) [inline package.json:23]: Big jump here. Any API changes that impact us?

**Review by mattmachell — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (Nicholassully) [reply package.json:50]: I made sure that the build worked and tests passed, with a few (not many) manual tests in the app

File: cypress/integration/filter.spec.js (hunk 2 above)
Comment (Nicholassully) [reply cypress/integration/filter.spec.js:19]: That's right is not testing the HTML but the behaviour. Maybe I could add a class/id with a prefix of `hook` to indicate it's only for a test?

File: package.json (hunk 3 above)
Comment (Nicholassully) [reply package.json:23]: I think that I was very lucky with this one, I tested that Cypress ran and got the correct outputs seemed to work for me.

File: cypress/integration/filter.spec.js (hunk 2 above)
Comment (mattmachell) [reply cypress/integration/filter.spec.js:19]: Potentially, but yeah if you have a label it should be for a uniquely identified element anyway (so assistive tech can associate them) which means you can us the for attribute on the label as a hook.

Looking on the Sirius dev env, I did notice an issue with those checkboxes though. HTML is:

`<input class="govuk-checkboxes__input task-type" id="Casework - Complaint review" name="selected-task-type" type="checkbox" value="CCR">
<label class="govuk-label govuk-checkboxes__label" for="">Casework - Complaint review</label>`

So it looks like id has got mixed up with a label or title attribute? For an ID I'd expect to see something like `select-task-checkbox-compliant-review` which was unique to that checkbox and was used in the label's for attribute, something like:

`  <input class="govuk-checkboxes__input task-type" **id="select-task-checkbox-compliant-review"** name="selected-task-type" type="checkbox" value="CCR">
<label class="govuk-label govuk-checkboxes__label" **for="select-task-checkbox-compliant-review"**>Casework - Complaint review</label>`

If you fixed that you'd then be able to use `label[for="select-task-checkbox-compliant-review"]` as your hook in the test.

**Comment by denialanderror** [comment]
My general approach when going up major versions (the first number in 1.2.3) is to look through the release notes. Major versions mean breaking changes, and well-maintained libraries will provide a migration guide. For example, [this is the migration guide for Cypress v8](https://docs.cypress.io/guides/references/migration-guide). If it is a small change, you will probably get lucky and not need to do anything, but it is worth still going through them to double check.

**Pull request closed without merging**